cmd/instance_pool_update.go

```diff
@@ -61,123 +60,160 @@ func (c *instancePoolUpdateCmd) cmdPreRun(cmd *cobra.Command, args []string) err
-		instancePool.IPv6Enabled = &c.IPv6
+		updateReq.Ipv6Enabled = &c.IPv6
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.Labels)) {
-		instancePool.Labels = &c.Labels
+		updateReq.Labels = convertIfSpecialEmptyMap(c.Labels)
+		updated = true
+	}
+
+	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.MinAvailable)) {
+		updateReq.MinAvailable = &c.MinAvailable
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.Name)) {
-		instancePool.Name = &c.Name
+		updateReq.Name = c.Name
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.PrivateNetworks)) {
-		privateNetworkIDs := make([]string, len(c.PrivateNetworks))
-		for i, v := range c.PrivateNetworks {
-			privateNetwork, err := globalstate.EgoscaleClient.FindPrivateNetwork(ctx, c.Zone, v)
+		updateReq.PrivateNetworks = make([]v3.PrivateNetwork, len(c.PrivateNetworks))
+		pn, err := client.ListPrivateNetworks(ctx)
+		if err != nil {
+			return fmt.Errorf("error listing Elastic IP: %w", err)
+		}
+		for i := range c.PrivateNetworks {
+			privateNetwork, err := pn.FindPrivateNetwork(c.PrivateNetworks[i])
 			if err != nil {
 				return fmt.Errorf("error retrieving Private Network: %w", err)
 			}
-			privateNetworkIDs[i] = *privateNetwork.ID
+			updateReq.PrivateNetworks[i] = v3.PrivateNetwork{ID: privateNetwork.ID}
 		}
-		instancePool.PrivateNetworkIDs = &privateNetworkIDs
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.SecurityGroups)) {
-		securityGroupIDs := make([]string, len(c.SecurityGroups))
-		for i, v := range c.SecurityGroups {
-			securityGroup, err := globalstate.EgoscaleClient.FindSecurityGroup(ctx, c.Zone, v)
+		sgs, err := client.ListSecurityGroups(ctx)
+
+		if err != nil {
+			return fmt.Errorf("error listing Security Group: %w", err)
+		}
+		updateReq.SecurityGroups = make([]v3.SecurityGroup, len(c.SecurityGroups))
+
+		for i := range c.SecurityGroups {
+			securityGroup, err := sgs.FindSecurityGroup(c.SecurityGroups[i])
 			if err != nil {
 				return fmt.Errorf("error retrieving Security Group: %w", err)
 			}
-			securityGroupIDs[i] = *securityGroup.ID
+			updateReq.SecurityGroups[i] = v3.SecurityGroup{ID: securityGroup.ID}
 		}
-		instancePool.SecurityGroupIDs = &securityGroupIDs
 		updated = true
 	}
 
-	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.InstanceType)) {
-		instanceType, err := globalstate.EgoscaleClient.FindInstanceType(ctx, c.Zone, c.InstanceType)
-		if err != nil {
-			return fmt.Errorf("error retrieving instance type: %w", err)
-		}
-		instancePool.InstanceTypeID = instanceType.ID
-		updated = true
-	}
+	//if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.InstanceType)) { //
+	//	instanceTypes, err := client.ListInstanceTypes(ctx)
+	//
+	//	if err != nil {
+	//		return fmt.Errorf("error listing instance type: %w", err)
+	//	}
+	//
+	//	updateReq.InstanceTypeID = instanceType.ID
+	//	updated = true
+	//}
```

Comment:
please don't leave commented code in PRs

Reply:
this was to be reviewed as I mentioned in the description, because I was not sure about it.
in case like this where I want idea from the review, should it be draft PR?